tidy3d/plugins/mode/mode_solver.py

```diff
@@ -317,27 +318,410 @@ def data_raw(self) -> ModeSolverData:
         if self.mode_spec.group_index_step > 0:
             return self._get_data_with_group_index()
 
-        # Compute data on the Yee grid
-        mode_solver_data = self._data_on_yee_grid()
+        if not self.mode_spec.bend_correction:
+            # Compute data on the Yee grid
+            mode_solver_data = self._data_on_yee_grid()
 
-        # Colocate to grid boundaries if requested
-        if self.colocate:
-            mode_solver_data = self._colocate_data(mode_solver_data=mode_solver_data)
+            # Colocate to grid boundaries if requested
+            if self.colocate:
+                mode_solver_data = self._colocate_data(mode_solver_data=mode_solver_data)
 
-        # normalize modes
-        self._normalize_modes(mode_solver_data=mode_solver_data)
+            # normalize modes
+            self._normalize_modes(mode_solver_data=mode_solver_data)
 
-        # filter polarization if requested
-        if self.mode_spec.filter_pol is not None:
-            self._filter_polarization(mode_solver_data=mode_solver_data)
+            # filter polarization if requested
+            if self.mode_spec.filter_pol is not None:
+                self._filter_polarization(mode_solver_data=mode_solver_data)
 
-        # sort modes if requested
-        if self.mode_spec.track_freq and len(self.freqs) > 1:
-            mode_solver_data = mode_solver_data.overlap_sort(self.mode_spec.track_freq)
+            # sort modes if requested
+            if self.mode_spec.track_freq and len(self.freqs) > 1:
+                mode_solver_data = mode_solver_data.overlap_sort(self.mode_spec.track_freq)
 
-        self._field_decay_warning(mode_solver_data.symmetry_expanded)
+            self._field_decay_warning(mode_solver_data.symmetry_expanded)
 
-        return mode_solver_data
+            return mode_solver_data
+        else:
+            # Create a reference solver with rotated geometries for interpolations
```

Comment:
Maybe slightly more readable would be to gather everything that is in this "else" into a `self.rotated_mode_data` method and have the logic here be something like

```
if self.mode_spec.bend_correction:
    return self.rotated_mode_data()

# then no change to the rest of the code since you don't need an ``else`` after a ``return``.
```